src/promptflow/tests/test_configs/flows/async_tools_with_sync_tools/async_passthrough.py

```diff
@@ -3,19 +3,19 @@
 
 
 @tool
-async def passthrough_str_and_wait(input1: str, wait_seconds=3) -> str:
+async def passthrough_str_and_wait(input1: str, wait_seconds=3, wait_seconds_in_cancellation=1) -> str:
     assert isinstance(input1, str), f"input1 should be a string, got {input1}"
     try:
         print(f"Wait for {wait_seconds} seconds in async function")
         for i in range(wait_seconds):
             print(i)
             await asyncio.sleep(1)
     except asyncio.CancelledError:
-        print("Async function is cancelled, start time consuming cancellation process")
-        import time
-        for i in range(3):
-            print(i)
-            time.sleep(1)
+        print(f"Async function is cancelled, wait for {wait_seconds_in_cancellation}"
+              " in cancellation process")
+        for i in range(wait_seconds_in_cancellation):
+            print(f"Wait for {i} seconds in async tool cancellation logic")
+            await asyncio.sleep(1)
```

Comment:
Just out of curiosity, seems like customer use `await asyncio.sleep(1)` or `time.sleep(1)` will not affect our cancellation.
We are creating a new thread to monitor the cancel in `monitor_coroutine_after_cancellation`.

Reply:
Good question, changed async sleep, so that the cancellation logic of tools won't block each other.
It should be a recommended practice just as we should avoid using blocker operation in tool's main logic.

But yes, it does not stop the flow from being cancelled.